discord/bot.py

```diff
@@ -532,14 +532,70 @@ def command(self, **kwargs):
         """
         return self.application_command(**kwargs)
 
-    def command_group(
-        self, name: str, description: str, guild_ids=None
+    def create_group(
+        self,
+        name: str,
+        description: str,
+        guild_ids: Optional[List[int]] = None,
     ) -> SlashCommandGroup:
-        # TODO: Write documentation for this. I'm not familiar enough with what this function does to do it myself.
+        """A shortcut method that creates a slash command group with no subcommands and adds it to the internal
+        command list via :meth:`~.ApplicationCommandMixin.add_application_command`.
+
+        .. versionadded:: 2.0
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The name of the group to create.
+        description: :class:`str`
+            The description of the group to create.
+        guild_ids: Optional[List[:class:`int`]]
+            A list of the IDs of each guild this group should be added to, making it a guild command.
+            This will be a global command if ``None`` is passed.
+
+        Returns
+        --------
+        SlashCommandGroup
+            The slash command group that was created.
+        """
         group = SlashCommandGroup(name, description, guild_ids)
         self.add_application_command(group)
         return group
 
+    def group(
+        self,
+        name: str,
+        description: str, 
+        guild_ids: Optional[List[int]] = None,
+    ) -> Callable[[Type[SlashCommandGroup]], SlashCommandGroup]:
+        """A shortcut decorator that initializes the provided subclass of :class:`.SlashCommandGroup`
+        and adds it to the internal command list via :meth:`~.ApplicationCommandMixin.add_application_command`.
+
+        .. versionadded:: 2.0
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The name of the group to create.
+        description: :class:`str`
+            The description of the group to create.
+        guild_ids: Optional[List[:class:`int`]]
+            A list of the IDs of each guild this group should be added to, making it a guild command.
+            This will be a global command if ``None`` is passed.
+
+        Returns
+        --------
+        Callable[[Type[SlashCommandGroup]], SlashCommandGroup]
+            The slash command group that was created.
+        """
+        def inner(cls: Type[SlashCommandGroup]) -> SlashCommandGroup:
+            group = cls(name, description, guild_ids=guild_ids)
+            self.add_application_command(group)
+            return group
```

Comment:
Maybe support classes that are not subclasses of SlashCommandGroup? (Untested)
```suggestion
        def inner(cls: Type[Any]) -> SlashCommandGroup:
            if not isinstance(cls, SlashCommandGroup):
                class cls(cls, SlashCommandGroup):
                    pass
            group = cls(name, description, guild_ids=guild_ids)
            self.add_application_command(group)
            return group
```

Reply:
This implementation doesn't seem to work and I also don't think we need this for 2.0.